api/auth.go

```diff
@@ -0,0 +1,23 @@
+package main
+
+import (
+	"github.com/gin-gonic/gin"
+)
+
+func Auth(requiredRole string) gin.HandlerFunc{
+	return func(context *gin.Context) {
+		tokenString := context.GetHeader("Authorization")
+		if tokenString == "" {
+			context.JSON(401, gin.H{"error": "request does not contain an access token"})
+			context.Abort()
+			return
+		}
+		err:= ValidateToken(tokenString, requiredRole)
```

Comment:
Where is this function

Reply:
I see it in `api/jwt.go` we should add the import

Reply:
moving stuff around still, probably all going to go into auth.go